stat/distuv/poisson.go

```diff
@@ -66,26 +66,49 @@ func (p Poisson) Prob(x float64) float64 {
 
 // Rand returns a random sample drawn from the distribution.
 func (p Poisson) Rand() float64 {
-	// poisson generator based upon the multiplication of
-	// uniform random variates.
-	// see:
-	//  Non-Uniform Random Variate Generation,
-	//  Luc Devroye (p504)
-	//  http://www.eirene.de/Devroye.pdf
-	x := 0.0
-	prod := 1.0
-	exp := math.Exp(-p.Lambda)
+	// NUMERICAL RECIPES IN C: THE ART OF SCIENTIFIC COMPUTING (ISBN 0-521-43108-5)
+	// p. 294
+	// <http://www.aip.de/groups/soe/local/numres/bookcpdf/c7-3.pdf>
+
+	g := math.Exp(-p.Lambda)
```

Comment:
Like the previous PR, this should work in log space. Specifically, there's no need for `g`, and then we can have `rnd := rand.ExpFloat64`, `t += rnd()`, `if t >= p.Lambda`

Reply:
Do you want the rejection method to work in log space, as well?